libs/s25main/lua/LuaInterfaceSettings.cpp

```diff
@@ -237,3 +237,15 @@ std::vector<AddonId> LuaInterfaceSettings::GetAllowedAddons()
     }
     return std::vector<AddonId>();
 }
+
+bool LuaInterfaceSettings::IsMapPreviewEnabled()
+{
+    kaguya::LuaRef getMapPreviewEnabled = lua["isMapPreviewEnabled"];
+    if(getMapPreviewEnabled.type() == LUA_TFUNCTION)
+    {
+        kaguya::LuaRef previewEnabled = getMapPreviewEnabled();
+        if(previewEnabled.typeTest<bool>())
+            return previewEnabled;
+    }
```

Comment:
Simpler version similar to https://github.com/Return-To-The-Roots/s25client/blob/605bd80f5cbb7c123dc5ba581124baae70ec783c/libs/s25main/lua/LuaInterfaceSettings.cpp#L180

```suggestion
        return getMapPreviewEnabled.call<bool>();
```

And maybe rename to `func` as there is already a naming-mix between "get" and "is"